File: js/src/maplestory/api/dto/characterLinkSkillDto.ts
import { CharacterLinkSkillInfoDto } from './characterLinkSkillInfoDto';
import { CharacterLinkSkillDtoBody } from '../response/characterLinkSkillDtoBody';

/**
 * 캐릭터 링크 스킬 정보
 */
class CharacterLinkSkillDto {
  /**
   * 조회 기준일
   */
  date: Date;

  /**
   * 캐릭터 직업
   */
  characterClass: string;

  /**
   * 링크 스킬 정보
   */
  characterLinkSkill: CharacterLinkSkillInfoDto;

  /**
   * 내 링크 스킬 정보
   */
  characterOwnedLinkSkill: CharacterLinkSkillInfoDto;

  constructor(obj: CharacterLinkSkillDtoBody) {
    const {
      date,
      character_class,
      character_link_skill,
      character_owned_link_skill,
    } = obj;

    this.date = new Date(date);
    this.characterClass = character_class;
    this.characterLinkSkill = new CharacterLinkSkillInfoDto(
      character_link_skill,
    );
    this.characterOwnedLinkSkill = new CharacterLinkSkillInfoDto(
      character_owned_link_skill,
    );
  }
}

export { CharacterLinkSkillDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/CharacterLinkSkillDTO.java
package dev.spiralmoon.maplestory.api.dto;

import com.google.gson.annotations.SerializedName;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.time.LocalDateTime;

/**
 * 캐릭터 링크 스킬 정보
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterLinkSkillDTO {

    /**
     * 조회 기준일 (KST)
     */
    @SerializedName("date")
    private String date;

    /**
     * 캐릭터 직업
     */
    @SerializedName("character_class")
    private String characterClass;

    /**
     * 링크 스킬 정보
     */
    @SerializedName("character_link_skill")
    private CharacterLinkSkillInfoDTO characterLinkSkill;

    /**
     * 내 링크 스킬 정보
     */
    @SerializedName("character_owned_link_skill")
    private CharacterLinkSkillInfoDTO characterOwnedLinkSkill;

    public LocalDateTime getDate() {
        return Utils.toLocalDateTime(this.date);
    }
}
